diff --git a/src/dmvfs.c b/src/dmvfs.c
index aa3dd5d..0732016 100644
--- a/src/dmvfs.c
+++ b/src/dmvfs.c
@@ -292,9 +292,23 @@ static Dmod_Context_t* find_fs_by_name(const char* fs_name)
     if(!Dmod_IsModuleLoaded(fs_name))
     {
         fs_context = Dmod_LoadModuleByName(fs_name);
+        if(fs_context == NULL)
+        {
+            DMOD_LOG_WARN("Module '%s' could not be loaded\n", fs_name);
+            unlock_mutex();
+            return NULL;
+        }
+        if(!Dmod_Enable(fs_context, false, NULL))
+        {
+            DMOD_LOG_WARN("Module '%s' could not be enabled\n", fs_name);
+            Dmod_UnloadModule(fs_name, true);
+            unlock_mutex();
+            return NULL;
+        }
         if(Dmod_GetDifFunction(fs_context, dmod_dmfsi_fopen_sig) == NULL)
         {
             DMOD_LOG_WARN("Module '%s' is not a valid file system\n", fs_name);
+            Dmod_Disable(fs_context, true);
             Dmod_UnloadModule(fs_name, true);
             fs_context = NULL;
         }
